cog/core/singleton.py

```
@@ -0,0 +1,8 @@
+class SingletonMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
```

Ruff (0.8.2)

Missing return type annotation for special method `__call__`

(ANN204)